src/fides/api/ctl/migrations/versions/643249f65453_case_insensitive_usernames_delete_.py

```diff
@@ -0,0 +1,55 @@
+"""case_insensitive_usernames_delete_duplicates
+
+Revision ID: 643249f65453
+Revises: 5d62bab40b71
+Create Date: 2023-02-01 22:22:38.055862
+
+"""
+from alembic import op
+from sqlalchemy.orm.session import Session
+from sqlalchemy.sql import text
+
+from fides.lib.models.fides_user import FidesUser
+from fides.lib.models.fides_user_permissions import FidesUserPermissions
+
+# revision identifiers, used by Alembic.
+revision = "643249f65453"
+down_revision = "5d62bab40b71"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+
+    session = Session(bind=op.get_bind())
+    conn = session.connection()
+    statement = text(
+        """SELECT DISTINCT ON (LOWER(username)) username, created_at, id
+FROM fidesuser
+ORDER BY LOWER(username), created_at ASC"""
+    )
+
+    result = conn.execute(statement)
+    original_user_ids = [row[2] for row in result]
+    op.execute(
+        f"UPDATE client SET user_id = NULL WHERE NOT(user_id=ANY(ARRAY[{original_user_ids}]::varchar[]))"
+    )
+    op.execute(
+        f"UPDATE auditlog SET user_id = NULL WHERE NOT(user_id=ANY(ARRAY[{original_user_ids}]::varchar[]))"
+    )
```

Comment:
I searched the codebase and these were the only other records that had a  `user_id` column

Reply:
Great.

One other suggestion, what do you think about replacing this user_id with something like "user_deleted" here? instead of leaving empty for the AuditLogs?

Reply:
I like that. I'll update it to "user_deleted"